### PR TITLE
Allow adding IBuildTrigger's to IBuildConfig

### PR DESCRIPTION
**Pull request by aslakknutsen**

### CODE REVIEW

File: src/main/java/com/openshift/internal/restclient/model/BuildConfig.java [hunk 1]
```diff
@@ -77,6 +79,38 @@ public BuildConfig(ModelNode node, IClient client, Map<String, String []> proper
...
+			triggerNode.get(new String[] {"generic", "secret"}).set(generic.getSecret());
+			break;
+		case github:
+			if(!(trigger instanceof IWebhookTrigger)) {
+				throw new IllegalArgumentException("IBuildTrigger of type github does not implement IWebhookTrigger");
+			}
+			IWebhookTrigger github = (IWebhookTrigger)trigger;
+			triggerNode.get(new String[] {"github", "secret"}).set(github.getSecret());
+			break;
+		case imageChange:
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: @aslakknutsen these work for now, but the back-end has moved to object refs which is simpler and only requires you to reference an instance of a kind by name.  Don't quote me, but I think its the name of an imagestream

**Comment by jcantrill** [comment]
@aslakknutsen LGTM.  I think it needs a rebase for me to merge

**Comment by aslakknutsen** [comment]
@jcantrill Branch rebased.

File: src/main/java/com/openshift/internal/restclient/model/BuildConfig.java (hunk 1 above)
Comment (aslakknutsen) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: @jcantrill Right, just mapping what was there. I'll look at a possible v1beta4? update.

**Comment by jcantrill** [comment]
cc @aslakknutsen  merged

**Pull request merged**